x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/comparison_side/utils.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import stringify from 'json-stable-stringify';
+import { Version } from '../versions_picker/constants';
+import type {
+  DiffableAllFields,
+  ThreeWayDiff,
+} from '../../../../../../../common/api/detection_engine';
+
+/**
+ * Picks the field value for a given version either from a three-way diff object or from a user-set resolved value.
+ *
+ * @param version - The version for which the field value is to be picked.
+ * @param fieldThreeWayDiff - The three-way diff object containing the field values for different versions.
+ * @param resolvedValue - The user-set resolved value resolved value. Used if it is set and the version is final.
+ * @returns - The field value for the specified version
+ */
+export function pickFieldValueForVersion<FieldName extends keyof DiffableAllFields>(
+  version: Version,
+  fieldThreeWayDiff: ThreeWayDiff<DiffableAllFields[FieldName]>,
+  resolvedValue?: DiffableAllFields[FieldName]
+): DiffableAllFields[FieldName] | undefined {
+  if (version === Version.Final) {
+    return resolvedValue ?? fieldThreeWayDiff.merged_version;
+  }
+
+  const versionFieldToPick = `${version}_version` as const;
+  return fieldThreeWayDiff[versionFieldToPick];
+}
+
+/**
+ * Stringifies a field value to an alphabetically sorted JSON string.
+ */
+export const stringifyToSortedJson = (fieldValue: unknown): string => {
+  if (fieldValue === undefined) {
+    return '';
+  }
+
+  if (typeof fieldValue === 'string') {
+    return fieldValue;
+  }
+
+  return stringify(fieldValue, { space: 2 });
+};
```

Comment:
nit: we define an identical function for the per-field diff [here](https://github.com/elastic/kibana/blob/5f49cc50178d4abc67be420534f8cfeafb60c954/x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/per_field_diff/get_field_diffs_for_grouped_fields.ts#L18), might be worth abstracting both of them out

Reply:
Yes, it's actually the same function. My idea was to just copy-paste it from the per-field diff, since we are going to remove the per-field diff tab and substitute it with this new tab once it's ready. But now I think it might be a good idea to abstract it away in a shared dir.

Reply:
Yes, it's actually the same function. I copy-pasted it from the per-field diff. I figured that copy-pasting is fine since we are going to remove the per-field diff code at some point once this new tab is ready. But since the function is very generic, perhaps it's a good idea to extract it away.

Any recommendations for a good place to put it? I couldn't find a suitable utils file. If you have a recommendation, I can move it there in the next PR.

Reply:
Not exactly the same thing but we could put it in this helpers file: `x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/helpers.ts`? I probably should've put it there initially as we have another similar function in `rule_diff_tab.tsx` that you originally added.

Reply:
Thanks! Will add to my todo for the next PR.